app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx

```diff
@@ -104,35 +105,52 @@ const PageRequestReview = ({
       [
         NimbusExperimentStatus.PREVIEW,
         NimbusExperimentStatus.DRAFT,
-        NimbusExperimentStatus.REVIEW,
-      ].map((status: NimbusExperimentStatus) => async () => {
-        try {
-          setSubmitError(null);
-
-          const result = await updateExperiment({
-            variables: {
-              input: {
-                id: currentExperiment.current!.id,
-                status,
+        NimbusExperimentPublishStatus.APPROVED,
+      ].map(
+        (
+          status: NimbusExperimentStatus | NimbusExperimentPublishStatus,
```

Comment:
This code generating handler functions is pretty code-golfed already, but maybe something like this would be a bit clearer and not require the conditional with `Object.keys` later on?
```javascript

      [
        { status: NimbusExperimentStatus.PREVIEW },
        { status: NimbusExperimentStatus.DRAFT },
        { publishStatus: NimbusExperimentPublishStatus.APPROVED },
      ].map((changes: Partial<ExperimentInput>) => async () => {
        /* ... */
          const result = await updateExperiment({
            variables: {
              input: {
                id: currentExperiment.current!.id,
                ...changes,
              },
            },
          });
        /* ... */
      }),
```

Reply:
Like this suggestion, updated